src/sentry/utils/kafka.py

```diff
@@ -1,11 +1,42 @@
+import datetime
 import logging
 import signal
+import time
+from typing import Optional
+
+from sentry import options
 
 logger = logging.getLogger(__name__)
 
 
-def run_processor_with_signals(processor):
+def delay_kafka_rebalance(configured_delay: int) -> None:
+    """
+    get current time.
+    get the seconds part of the current time.
+
+    get the seconds delay to sleep from an option.
+
+    and then continue to sleep for 0.5 seconds until you get to the exact second when you want to start/stop the app.
+    """
+    now = float(datetime.now().strftime("%S.%f"))
+
+    next_tick, remainder = divmod(now, configured_delay)
+    if remainder > 0:
+        next_tick += 1
+
+    seconds_sleep = (configured_delay * next_tick) - now
+    while seconds_sleep >= 0.5:
```

Comment:
There's an assumption here that the "error margin" of 0.5 seconds is acceptable, i.e. even if we're not exactly at the 15-second marks of the minute, it's ok as long as the consumers are synced (which is the main goal)

Reply:
You are right. You can simplify this by sleeping for the exact time of the next tick.